src/test/java/org/json/junit/JSONArrayTest.java

```diff
@@ -66,6 +61,194 @@ public class JSONArrayTest {
+        String actual = (String)firstString.get();
+        assertEquals("abc", actual);
+    }
+
+    /**
+     * 1.2.1 Stream JSONArray through a map
+     */
+    @Test
+    public void testStream_1_2_1() {
+        String data = "[ 1, 2, 3, true, false, null, \"abc\", \"def\", { \"a\": \"b\" }, []]";
+        JSONArray jsonArray = new JSONArray(data);
+        Object[] expectedList = { "Number: 1", "Number: 2", "Number: 3" };
+
+        List<String> actualList = jsonArray.stream()
+                .filter(x -> x instanceof Number)
+                .map(x -> "Number: " + x)
+                .collect(Collectors.toList());
+
+        assertEquals(expectedList.length, actualList.size());
+        for (int i = 0; i < actualList.size(); ++i) {
+            assertEquals(expectedList[i], actualList.get(i));
+        }
+    }
+
+    /**
+     * 1.2.2 Stream JSONArray to find min and max
+     */
+    @Test
+    public void testStream_1_2_2() {
+        String data = "[ 1, 2, 3, true, false, null, \"abc\", \"def\", { \"a\": \"b\" }, []]";
+        JSONArray jsonArray = new JSONArray(data);
+
+        Optional<Object> min = jsonArray.stream()
+                .filter(x -> x instanceof Number)
+                .min(Comparator.comparingInt(x -> (int) x));
+        Optional<Object> max = jsonArray.stream()
+                .filter(x -> x instanceof Number)
+                .max(Comparator.comparingInt(x -> (int) x));
+
+        // Assert
+        assertTrue(min.isPresent());
+        assertEquals(1, min.get());
+
+        assertTrue(max.isPresent());
+        assertEquals(3, max.get());
+    }
+
+    /**
+     * 1.2.3 Stream JSONArray through groupingBy
+     */
+    @Test
+    public void testStream_1_2_3() {
+        String data = "[ 1, 2, 3, true, false, null, \"abc\", \"def\", { \"a\": \"b\" }, []]";
+        JSONArray jsonArray = new JSONArray(data);
+        Object[] expectedList = { "Number: 1", "Number: 2", "Number: 3" };
+
+        Map<Object, List<Object>> groupedByMod2 = jsonArray.stream()
+                .filter(x -> x instanceof Number)
+                .collect(Collectors.groupingBy(x -> (int) x % 2));
+
+        // Assert
+        assertEquals(2, groupedByMod2.size());
+
+        assertEquals(1, groupedByMod2.get(0).size());
+        assertEquals(2, groupedByMod2.get(0).get(0));
+
+        assertEquals(2, groupedByMod2.get(1).size());
+        assertEquals(1, groupedByMod2.get(1).get(0));
+        assertEquals(3, groupedByMod2.get(1).get(1));
+    }
+
+    /**
+     * 1.3.1 Stream empty JSONArray
+     */
+    @Test
+    public void testStream_1_3_1() {
+        JSONArray jsonArray = new JSONArray();
+
+        List<Object> list = jsonArray.stream()
+                .collect(Collectors.toList());
+
+        assertTrue(list.isEmpty());
+    }
+
+    /**
+     * 1.3.2 Stream null JSONArray
+     */
+    @Test
+    public void testStream_1_3_2() {
+        JSONArray jsonArray = null;
+
+        assertThrows(NullPointerException.class, () -> {
```

Comment:
I don't think this test is adding anything here. It seems obvious that an NPE would be thrown in this case.

Reply:
Removed